src/link/index.ts

```diff
@@ -246,13 +250,29 @@ async function linkNewPackages (
   await Promise.all([
     linkAllModules(newPkgs, pkgsToLink, {optional: opts.optional}),
     (async () => {
-      try {
-        await linkAllPkgs(linkPkg, newPkgs, opts)
-      } catch (err) {
-        if (!err.message.startsWith('EXDEV: cross-device link not permitted')) throw err
-        logger.warn(err.message)
-        logger.info('Falling back to copying packages from store')
-        await linkAllPkgs(copyPkg, newPkgs, opts)
+      // this works in the following way:
+      // - hardlink: hardlink the packages, no fallback
+      // - reflink: reflink the packages, no fallback
+      // - auto: try to hardlink the packages, if it fails, fallback to copy
+      // - copy: copy the packages, do not try to link them first
+      switch (opts.packageImportMethod) {
+        case 'reflink':
+          await linkAllPkgs(reflinkPkg, newPkgs, opts)
+          break
+        case 'hardlink':
+          await linkAllPkgs(hardlinkPkg, newPkgs, opts)
+          break
+        case 'auto':
+          try {
+            await linkAllPkgs(hardlinkPkg, newPkgs, opts)
+            break
+          } catch (err) {
+            if (!err.message.startsWith('EXDEV: cross-device link not permitted')) throw err
+            logger.warn(err.message)
+            logger.info('Falling back to copying packages from store')
+          }
```

Comment:
I did not configure linting yet but it wouldn't allow a case w/o break, so better just duplicate the copy code here and use a break